feat(xorBy, xorWith): add `xorBy`, `xorWith` to `compat/array`

oror-sine opened the pull request:
close #996
close #997 

# test
<img width="676" alt="image" src="https://github.com/user-attachments/assets/a3b86b86-a40f-4efb-857b-d7e432f4ee36" />
<img width="674" alt="image" src="https://github.com/user-attachments/assets/b0c18943-9616-45f2-8bff-f7f79cf62807" />


# bench
<img width="809" alt="image" src="https://github.com/user-attachments/assets/09c5f0b2-cdad-42de-bce3-e1dbebf74795" />

oror-sine commented:
> Missing export in src/compat/index.ts: `export { xorBy } from './array/xorBy.ts';`

Thank you!

oror-sine commented:
I refactored the code to improve readability.

<details><summary>Code Changes</summary>
<p>

# Before
```typescript
export function xorBy<T>(...values: Array<ArrayLike<T> | null | undefined | Mapper<T>>): T[] {
  const lastValue = last(values);
  const arrays = values.filter(isArrayLikeObject).map(toArray);
  const flatten = flattenArrayLike(arrays.map(uniq));
  const mapper = isArrayLikeObject(lastValue) || lastValue == null ? identity : iteratee(lastValue);

  const itemMap = new Map();

  for (let i = 0; i < flatten.length; ++i) {
    const mapped = mapper(flatten[i]);
    if (!itemMap.has(mapped)) {
      itemMap.set(mapped, flatten[i]);
    } else {
      itemMap.set(mapped, null);
    }
  }

  const result = [];

  for (const item of itemMap.values()) {
    if (item != null) {
      result.push(item);
    }
  }

  return result;
}
```

# After
```typescript
export function xorBy<T>(...values: Array<ArrayLike<T> | null | undefined | Mapper<T>>): T[] {
  const lastValue = last(values);
  const arrays = values.filter(isArrayLikeObject).map(toArray).map(uniq);
  const flatten = flattenArrayLike<T>(arrays);
  const mapper = isArrayLikeObject(lastValue) || lastValue == null ? identity : iteratee(lastValue);

  const union = uniqBy(flatten, mapper);
  let intersection = Array<T>();

  for (let i = 0; i < arrays.length; ++i) {
    for (let j = i + 1; j < arrays.length; ++j) {
      intersection = unionBy(intersection, intersectionBy(arrays[i], arrays[j], mapper), mapper);
    }
  }

  return differenceBy(union, intersection, mapper) as T[];
}
```
</p>
</details> 

<details><summary>Test and Bench</summary>
<p>

# test
<img width="673" alt="image" src="https://github.com/user-attachments/assets/638c33bf-0a6c-45b7-9e0e-145f5ddb952a" />
<img width="671" alt="image" src="https://github.com/user-attachments/assets/3abdc169-a799-4fbf-bd3b-2f9f1d9790ec" />

# bench
<img width="674" alt="image" src="https://github.com/user-attachments/assets/d2556ecb-6419-45c7-b797-9924a73d0f9f" />
</p>
</details>

hsskey commented:
Mind checking this out?
https://github.com/toss/es-toolkit/wiki/es%E2%80%90toolkit-compat%EC%97%90-%EA%B8%B0%EC%97%AC%ED%95%98%EA%B8%B0

oror-sine commented:
I've just added `xorWith` to this PR, since it's closely related to `xorBy` and follows the same pattern.  
I’ve updated the PR description to reflect this change, including new tests and benchmarks.

The last commit only updated the description string inside a `describe` block in the tests — no logic or assertions were changed.  
However, CircleCI is currently failing. Could you please take a look when you have a moment?

raon0211 reviewed (approved):
Thanks for your contribution! 🙌
